runtime/vm/VMAccess.cpp

```diff
@@ -1010,6 +1024,11 @@ waitForExclusiveVMAccessMetronome(J9VMThread * vmThread, UDATA responsesRequired
 void
 waitForExclusiveVMAccessMetronomeTemp(J9VMThread * vmThread, UDATA vmResponsesRequired, UDATA jniResponsesRequired)
 {
+	J9JavaVM *vm = vmThread->javaVM;
+
+	/* should not try to wait since we should have not requested it in the first place */
+	Assert_VM_true(FALSE == vm->alreadyHaveExclusive);
```

Comment:
In #18535, the code returned if exclusive already had been acquired. Can `waitForExclusiveVMAccessMetronomeTemp` be called from the below code path and trigger the assertion? Will it be safer to return instead of asserting?
https://github.com/eclipse-openj9/openj9/blob/c2585deab318d526a08463389b0bbc28ab1ed636/runtime/gc_realtime/Scheduler.cpp#L463-L472

Reply:
Below is an improvement if we decide to keep the assertion.
```suggestion
	Assert_VM_false(vm->alreadyHaveExclusive);
```

Reply:
While semantically I still think the assert is more correct, the fact that shutdown path ignores the return result of requestExclusiveVMAccess perhaps may indeed lead to the assert triggering (in which case the shutdown path should be fixed), atm probably it is just simpler to relax the assert.

Reply:
The HCR operations are done through JVMTI functions. Shutdown can be requested through signal handling and JNI functions asynchronously, which might happen at the same time as HCR. In such a case, I am not completely sure if there is sufficient synchronization to hold shutdown until all threads have completed their work.